### PR TITLE
build: Build and deploy testnet

### PR DESCRIPTION
**Pull request by bgins**
### Review Type Requested (choose one):
- [ ] **Glance** - superficial check (from domain experts)
- [x] **Logic** - thorough check (from everybody doing review)

### Summary

This pull request implements the following changes:

- [x] Remove Cloudflare token and Doppler environment from solver and job creator Dockerfiles
- [x] Add `testnet_deploy_contracts` action
- [x] Add `testnet_deploy_services` action to build and deploy solver and job-creator to Testnet
- [x] Add `testnet_restart_solver` and `testnet_restart_job_creator` actions to restarts services individually
- [x] Update Devnet documentation to use Ubuntu instead of Amazon Linux

All GitHub actions are run on-demand for the moment.

Note that `testnet_deploy_contracts` depends on the #122 where the `arbitrumSepolia` network is defined. We could wait on this one if we prefer to hold off until the other PR is merged.

### Task/Issue reference

Implements https://github.com/Lilypad-Tech/lilypad/issues/124

### How to test this code?

Local builds for the Docker images should succeed and work with the local stack.

The GH actions can only be run on-demand once this PR reaches `main`, but here are the last runs while working on them:

- `testnet_deploy_services`: https://github.com/Lilypad-Tech/lilypad/actions/runs/9310164136
- `testnet_restart_solver`: https://github.com/Lilypad-Tech/lilypad/actions/runs/9312184243
- `testnet_restart_job_creator`: https://github.com/Lilypad-Tech/lilypad/actions/runs/9312184244

`testnet_deploy_contracts` has not been tested, but is quite similar to the Devnet contract deploy action: https://github.com/Lilypad-Tech/lilypad/blob/acf81d14097b7388c6a70ed230837b0a75363c6a/.github/workflows/devnet_deploy.yml#L9-L28

### CODE REVIEW

File: .github/workflows/testnet_deploy_contracts.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        uses: dopplerhq/cli-action@v1
+
+      - name: Deploy contracts
+        id: deploy-contracts
+        env:
+          DOPPLER_TOKEN: ${{ secrets.TESTNET_DOPPLER_TOKEN_CONTRACTS_DEPLOY }}
+        run: |
+          cd hardhat
+          npm ci
+          doppler run -- npx hardhat deploy --network arbitrumSepolia
```
Comment (AquiGorka) [inline .github/workflows/testnet_deploy_contracts.yml:24]: We need to validate this works locally, to confirm we get the same contract addresses every time.

File: .github/workflows/testnet_deploy_services.yml [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        uses: aws-actions/amazon-ecr-login@v1
+        with:
+          mask-password: 'true'
+
+      - name: Solver build, tag, and push image to Amazon ECR
+        id: build-image
+        env:
+          ECR_REGISTRY: ${{ steps.login-ecr.outputs.registry }}
+          ECR_REPOSITORY_SOLVER: ${{ secrets.ECR_REPOSITORY_SOLVER }}
+          CLOUDFLARE_TOKEN_SOLVER: ${{ secrets.TESTNET_CLOUDFLARE_TOKEN_SOLVER }}
```
Comment (AquiGorka) [inline .github/workflows/testnet_deploy_services.yml]: This one is no longer needed here at build time.

File: .github/workflows/testnet_deploy_services.yml [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+        with:
+          host: ${{ secrets.TESTNET_EC2_HOST_SOLVER }}
+          username: ${{ secrets.TESTNET_EC2_USERNAME_SOLVER }}
+          key: ${{ secrets.TESTNET_EC2_PRIVATE_KEY_SOLVER }}
+          envs: ECR_REGISTRY, ECR_REPOSITORY_SOLVER, TESTNET_DOPPLER_TOKEN_SOLVER
+          script_stop: true
+          script: |
+            docker stop solver || true
+            docker rm solver || true
+            aws ecr get-login-password --region us-east-1 | docker login --username AWS --password-stdin $ECR_REGISTRY
```
Comment (AquiGorka) [inline .github/workflows/testnet_deploy_services.yml:56]: If the ubuntu instance has the ecr access I believe this cmd can be removed, but let's double check.

File: .github/workflows/testnet_deploy_services.yml [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+        with:
+          host: ${{ secrets.TESTNET_EC2_HOST_JOB_CREATOR }}
+          username: ${{ secrets.TESTNET_EC2_USERNAME_JOB_CREATOR }}
+          key: ${{ secrets.TESTNET_EC2_PRIVATE_KEY_JOB_CREATOR }}
+          envs: ECR_REGISTRY, ECR_REPOSITORY_JOB_CREATOR, TESTNET_DOPPLER_TOKEN_JOB_CREATOR
+          script_stop: true
+          script: |
+            docker stop job-creator || true
+            docker rm job-creator || true
+            aws ecr get-login-password --region us-east-1 | docker login --username AWS --password-stdin $ECR_REGISTRY
```
Comment (AquiGorka) [inline .github/workflows/testnet_deploy_services.yml:116]: Same as above, let's see if this cmd is still needded.

File: .github/workflows/testnet_restart_job_creator.yml [hunk 5]
```diff
@@ -0,0 +1,17 @@
+name: Restart Testnet job creator
```
Comment (AquiGorka) [inline .github/workflows/testnet_restart_job_creator.yml:1]: 😍

File: DEVNET.md [hunk 6]
```diff
@@ -49,18 +49,39 @@ These steps have been used for [AWS](https://aws.amazon.com/) so maybe some chan
...
+
+The `ubuntu` user assumes the default user on Ubuntu images. The `newgrrp` command adds your user to the group without logging out and back in.
+
+Check that Docker is running:
+
+```sh
+docker run hello-world
 ```
 
 To grant access to the [ECR](https://aws.amazon.com/ecr/) registry from the VM execute `aws configure` with credentials that have `pull` permissions.
```
Comment (AquiGorka) [inline DEVNET.md]: Can you add a line referencing how the instance has to be added the role (and which role) to have ecr pull access?

**Review by AquiGorka — APPROVED** [review]
💃 🕺 🎉  thank you @bgins !

File: .github/workflows/testnet_deploy_services.yml (hunk 3 above)
Comment (bgins) [reply .github/workflows/testnet_deploy_services.yml:56]: This part is still needed for the integration with the Docker CLI.

Comment (bgins) [reply .github/workflows/testnet_deploy_services.yml:56]: But worth mentioning, the ECR access does allow us to make the `aws ecr get-login-password` call.

File: .github/workflows/testnet_deploy_services.yml (hunk 2 above)
Comment (bgins) [reply .github/workflows/testnet_deploy_services.yml]: Yup! Removed it 👌

File: DEVNET.md (hunk 6 above)
Comment (bgins) [reply DEVNET.md]: Yep! Added some details here: https://github.com/Lilypad-Tech/lilypad/blob/42504b47d8d7d78dcb1afda7ca6ecd62a4e419b8/DEVNET.md?plain=1#L54

File: .github/workflows/testnet_deploy_services.yml (hunk 3 above)
Comment (AquiGorka) [reply .github/workflows/testnet_deploy_services.yml:56]: Ahh, nice, with the role in place, we don't need to manually do `aws configure` when we provision the instance 👌

**Pull request merged**